timestamp-impl/src/main/java/com/palantir/timestamp/TimestampRanges.java

```diff
@@ -61,4 +53,37 @@ public static OptionalLong getTimestampMatchingModulus(TimestampRange range, int
                 ? OptionalLong.of(candidate)
                 : OptionalLong.empty();
     }
+
+    /**
+     * Returns the highest timestamp in the provided timestamp range that has the provided residue class modulo the
+     * provided modulus.
+     *
+     * If the timestamp range does not contain a timestamp matching the criteria, returns empty.
```

Comment:
i think this feels weird from a contract perspective. We never call this method in a way where it could return empty... this validation is all essentially redundant.

Maybe just have this method actually return the response object? like something that returns the combination of (base, count, multiple)?